src/Searchbar.vue

```diff
@@ -19,14 +19,19 @@ export default {
         return undefined;
       }
       const matches = [];
-      const regex = new RegExp(this.value, 'i');
+      const regexes = this.value.split(' ').map(searchKeyword => new RegExp(searchKeyword, 'i'));
       this.data.forEach((entry) => {
         const { headings, src, title } = entry;
         const keywords = entry.keywords || '';
-        let hasMatchingHeading = false;
+
+        let searchTarget = [title].concat(keywords).join(' ');
+        if (regexes.every(regex => regex.test(searchTarget))) {
+          matches.push(entry);
+        }
+
         Object.entries(headings).forEach(([id, text]) => {
-          if (regex.test(text)) {
-            hasMatchingHeading = true;
+          searchTarget = [title].concat(keywords).concat(text).join(' ');
+          if (regexes.every(regex => regex.test(searchTarget))) {
```

Comment:
We can check `isMatchingPage` here.

```js
if (isMatchingPage || ...
```

Reply:
Sorry there's a bit of confusion here, do you mean `if (!isMatchingPage || ...` (notice the exclaimation mark), or is this a suggestion for optimizing the matching algorithm?

If the title or page keyword matches the search keywords, then we will display all headings that the page has, regardless of whether the heading contains those search keywords or not. This is an intentional design that Prof suggested.

Reply:
Optimizing by short-circuiting.